jobs/build.Jenkinsfile

```diff
@@ -247,84 +258,93 @@ lock(resource: "build-${params.STREAM}") {
-                    def meta = readJSON(text: shwrapCapture("cosa meta --build=${buildID} --dump"))
-                    def rev = meta["coreos-assembler.config-gitrev"]
-                    currentBuild.description = "${build_description} 🔨 ${buildID}"
-                    // Run the mArch jobs and wait. We wait here because if they fail
-                    // we don't want to bother running the release job again since the
-                    // goal is to get a complete build.
-                    run_multiarch_jobs(missing_arches, rev, buildID, cosa_img, true)
-                    if (stream_info.type != "production") {
-                        run_release_job(buildID)
+            def buildID = shwrapCapture("readlink builds/latest")
+            if (prevBuildID == buildID) {
+                currentBuild.result = 'SUCCESS'
+                currentBuild.description = "${build_description} 💤 (no new build)"
+
+                // Nothing changed since the latest build. Check if it's missing
+                // some arches and retrigger `build-arch` only for the missing
+                // arches, and the follow-up `release` job. Match the exact src
+                // config commit that was used. But only do this if there isn't
+                // already outstanding work in progress for that build ID. Skip if
+                // not uploading since it's required for multi-arch.
+                if (uploading && !buildid_has_work_pending(buildID, additional_arches)) {
+                    def builds = readJSON file: "builds/builds.json"
+                    assert buildID == builds.builds[0].id
+                    def missing_arches = additional_arches - builds.builds[0].arches
+                    if (missing_arches) {
+                        def meta = readJSON(text: shwrapCapture("cosa meta --build=${buildID} --dump"))
+                        def rev = meta["coreos-assembler.config-gitrev"]
+                        currentBuild.description = "${build_description} 🔨 ${buildID}"
+                        // Run the mArch jobs and wait. We wait here because if they fail
+                        // we don't want to bother running the release job again since the
+                        // goal is to get a complete build.
+                        run_multiarch_jobs(missing_arches, rev, buildID, cosa_img, true)
+                        if (stream_info.type != "production") {
+                            run_release_job(buildID)
+                        }
                     }
                 }
-            }
 
-            // And we're done!
-            return
+                // And we're done!
+                return
+            }
+        newBuildID = buildID
+        } else{
+            stage("Import OCI image"){
+                shwrap("""
+                    cosa import docker://${source_oci_image}
+                """)
+                newBuildID = shwrapCapture("basename \$(realpath builds/latest)")
```

Comment:
let's pull the buildID from the builds.json: 

https://github.com/coreos/fedora-coreos-pipeline/blob/8b5f5f4bc563b5f6ae0e44d39d79eb8184f5d358/jobs/build.Jenkinsfile#L306-L307